src/core/lib/iomgr/tcp_server_posix.c

```diff
@@ -70,6 +71,8 @@
 
 #define MIN_SAFE_ACCEPT_QUEUE_SIZE 100
 
+int grpc_tcp_server_posix_expand_wildcard_addr = 0;
```

Comment:
The hardware probing happens in some external C++ vendor-specific code, so I wanted a flag that the vendor-specific code could set. This keeps the interface in the core C level very simple because it just reads the flag.

The ProbeHardware() function could be a callback set by vendor-specific code, but then it's still not a channel arg.